### ISSUE TITLE
Notes list causing frame drops

### ISSUE THREAD

**Issue by SnoFox** [kalilistic/NeatNoter#19]
When NeatNoter is left open on the list view, it causes frame drops/hitching akin to microstutter every 5 seconds. If you minimize the NeatNoter window to just be the titlebar, click into a note, or close the window completely, it alleviates the issue.

Example of it happening: https://youtu.be/wcFclX1SE0A (Minor spoilers. Sitting in Endwalker city, note titles show older dungeon names)

In linked video, you can see the players freeze as well as the ImGui showing plugin RAM usage at the following timestamps: 1s, 6s, 11s, 16s, 21s, 41s, 46s, 51s

I do have search set to include note contents, and a secondary backup dir on my local SSD. I've noticed this seems to auto-save every five seconds, even when I've made no changes.

I can attach my notes export if you believe it's note content causing the issue. I've categorized nearly every note, but only having 30 of them.

I've not yet attempted to see if it's a conflicting plugin; will be willing to test if you're unable to replicate/have no suspect.

Version info: 
* Dalamud vD6.2.0.16
* NeatNoter v2.1.2.0

Other loaded plugins: 
<details><summary>Plugin list</summary>

```
[22:56]Dalamud vD6.2.0.16 loaded. 47 plugin(s) loaded.
[22:56]    》 AntiAfkKick-Dalamud v2.1.0.0 loaded.
[22:56]    》 AutoVisor v1.2.1.1 loaded.
[22:56]    》 Character Sync v2.0.0.4 loaded.
[22:56]    》 Chat Bubbles v1.3.2.2 loaded.
[22:56]    》 DalamudRepoBrowser v1.0.2.1 loaded.
[22:56]    》 Damage Info v1.3.8.2 loaded.
[22:56]    》 Discord Rich Presence v2.0.0.0 loaded.
[22:56]    》 ezMiniCactpot v1.1.0.2 loaded.
[22:56]    》 ezWondrousTails v2.1.0.3 loaded.
[22:56]    》 Fader Plugin v1.0.2.2 loaded.
[22:56]    》 FC Name Color v2.0.2.3 loaded.
[22:56]    》 FlyTextFilter v3.3.0.0 loaded.
[22:56]    》 FPS Plugin v1.5.0.1 loaded.
[22:56]    》 GatherBuddy v2.2.1.4 loaded.
[22:56]    》 Glamaholic v1.9.2.0 loaded.
[22:56]    》 Globetrotter v1.2.4 loaded.
[22:56]    》 Good Memory v1.0.10 loaded.
[22:56]    》 Inventory Tools v1.0.0.7 loaded.
[22:56]    》 Item Search v1.8.2.2 loaded.
[22:56]    》 Kapture v1.3.1.0 loaded.
[22:56]    》 Macro Chain v2.1.0.1 loaded.
[22:56]    》 Market Board plugin v0.5.3.0 loaded.
[22:56]    》 NeatNoter v2.1.2.0 loaded.
[22:56]    》 No Kill Plugin v1.0.3.1 loaded.
[22:56]    》 No Tank You v2.3.1.2 loaded.
[22:56]    》 NoSoliciting v3.0.2 loaded.
[22:56]    》 Ocean Fishin' v2.2.0.0 loaded.
[22:56]    》 Orchestrion v1.6.0.3 loaded.
[22:56]    》 PaissaHouse v1.2.1.0 loaded.
[22:56]    》 Pat Me v1.2.0.1 loaded.
[22:56]    》 Penny Pincher v1.4.0.2 loaded.
[22:56]    》 PingPlugin v1.15.1.0 loaded.
[22:56]    》 Pixel Perfect v1.4.0.2 loaded.
[22:56]    》 PriceCheck v2.1.0.0 loaded.
[22:56]    》 QoL Bar v2.2.1.2 loaded.
[22:56]    》 Remind Me v1.1.0.0 loaded.
[22:56]    》 RepairMe v1.0.0.4 loaded.
[22:56]    》 Reset enmity command v0.0.1.1 loaded.
[22:56]    》 RezPls v1.2.1.1 loaded.
[22:56]    》 Simple Tweaks v1.6.6.0 loaded.
[22:56]    》 Teleporter v2.0.1.4 loaded.
[22:56]    》 The Great Separator v1.2.2.0 loaded.
[22:56]    》 Tourist v1.2.5 loaded.
[22:56]    》 Visibility v1.1.1.2 loaded.
[22:56]    》 WOLdo v1.3.0.1 loaded.
[22:56]    》 Wotsit v1.0.1.3 loaded.
[22:56]    》 Yes Already v1.2.0.17 loaded.
```
</details>

**Comment by kalilistic** [comment]
Thank you for the very detailed report! I have some ideas on the fix I'll look into.

**Comment by SnoFox** [comment]
Thanks for looking into this.

I've noticed today that the category list is also causing this issue. I had a note up and left the category picker behind it, which also was causing lag spikes. I simply closed the category picker and the issue was again mitigated.

**Comment by kalilistic** [comment]
I wasn't able to replicate this but I think it's because I was actually saving everything every 5 seconds. I made the saving smarter to only save the modified documents. I also added some options under settings to configure how frequently in performs the smaller and full saves. I've pushed it to my custom repo and it's pending a merge on the official testing repo. Let me know how it works out. https://plugins.kalilistic.io/
